### ISSUE TITLE
Design OCI deployment container interface

### ISSUE THREAD

**Issue by christoferlof** [microsoft/AzureTRE#42]
The Composition delegates responsibility of the CRUD operations on Workspaces and Workspace Services to a OCI deployment container. To be able to pass parameters and get back results with information about the Workspace Services provisioned an interface/contract must be defined.

- [ ] Describe how an action is passed to the OCI Deployment Container e.g. Install, Delete...
- [ ] Describe a structured approach to return Workspace Service information (e.g. VM address, IP..)
- [ ] Describe a way to return execution success/failed 
- [ ] Document the interface with examples

**Comment by marrobi** [comment]
@christoferlof we have an ADO story around template store. That requires an ADR, one of the options would be OCI Artifacts. 

Think this needs to go on hold until a template store ADR has been completed. I like OCI as an option but we need to be clear on what user experience will be.

**Comment by christoferlof** [comment]
I agree @marrobi. The authoring experience around workspaces services is important. Regarding the actual deployment requests, we still need an interface for the deployment client container.

**Comment by marrobi** [comment]
Firstly given this isn't be"prior art". We should give it some more thought.

For example, if the templates are Terraform, does it not make sense to use a private Terraform registry?

If ARM Templates, does Template Specs make sense?

Given the deployment runner is going to be IaC specific, do we need a single interface? Stick with the technologies likely to be used by the user?

What's wrong with checking out a git repo. As CI runners do?

As above, I like OCI Artifacts, as can bundle additional content, but think we need to answer some of these questions before settling on it as a solution.

**Comment by marrobi** [comment]
@christoferlof Apologies, am I getting confused between artifacts and the deployment runner?

The reference to workspace/workspace service is throwing me.

**Comment by christoferlof** [comment]
The proposed container contact was created but the scope of it and naming etc have changed so closing this issue in favor of a new
